actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -144,17 +144,21 @@ def initialize(parameters = {})
     end
 
     # Returns true if another +Parameters+ object contains the same content and
-    # permitted flag, or other Hash-like object contains the same content. This
-    # override is in place so you can perform a comparison with `Hash`.
-    def ==(other_hash)
-      if other_hash.respond_to?(:permitted?)
-        super
+    # permitted flag.
+    def ==(other)
+      if other.respond_to?(:permitted?)
+        self.permitted? == other.permitted? && self.parameters == other.parameters
```

Comment:
Yeah, I think I got the `permitted` and `parameters` ivars mixed up for a moment there.

I’m still not sure what you think is missing. I was considering adding this test case as well:

```rb
test "is equal to Parameters instance with same different source params, but same permitted params" do
  params1 = ActionController::Parameters.new(a: 1, b: 2).permit(:a)
  params2 = ActionController::Parameters.new(a: 1, c: 3).permit(:a)
  assert(params1 == params2)
end
```

If I change `self.permitted? == other.permitted? && self.parameters == other.parameters` to `self.to_h == other.to_h`, all tests (including the above) still pass. I would like to see an example that would pass when using `to_h` but fail when comparing `permitted?` and `parameters`.